tests/templates/test_layers/test_gate_fabric.py

```diff
@@ -0,0 +1,778 @@
+                        0.70710678 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.70710678 + 0.0j,
+                        0.0 + 0.0j,
+                    ]
+                ),
+            ),
+            (
+                qml.math.array([0, 1, 1, 1]),
+                qml.math.array(
+                    [
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.70710678 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.70710678 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                    ]
+                ),
+            ),
+            (
+                qml.math.array([1, 1, 0, 1]),
+                qml.math.array(
+                    [
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        -0.70710678 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.70710678 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                    ]
+                ),
+            ),
+            (
+                qml.math.array([1, 1, 1, 1]),
+                qml.math.array(
+                    [
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        0.0 + 0.0j,
+                        1.0 + 0.0j,
+                    ]
+                ),
+            ),
+        ],
+    )
+    def test_decomposition_q(self, init_state, exp_state, tol):
+        """Test the decomposition of the Q_{theta, phi}` exchange gate by asserting the prepared
+        state."""
+
+        N = 4
+        wires = range(N)
+
+        weight = [[[np.pi / 2, np.pi / 2]]]
+
+        dev = qml.device("default.qubit", wires=N)
+        print(init_state, exp_state)
```

Comment:
This seems like a leftover from local testing.

Reply:
Removed.